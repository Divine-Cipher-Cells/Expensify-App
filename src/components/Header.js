import React from "react";
import { NavLink } from "react-router-dom";
const Header = (props) =>(
    <header>
    <h1>{props.children}</h1>
    <NavLink to="/" activeClassName="is-active" exact strict >Dashboard </NavLink>
    <NavLink to="/create" activeClassName="is-active" exact strict>Create-Expense </NavLink>
    </header>
);
export default Header;