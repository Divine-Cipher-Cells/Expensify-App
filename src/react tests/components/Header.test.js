import React from "react";
import Header from "../../components/Header";
import {shallow} from "enzyme";



test("shallow test for Header component", () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot(); //this statememnt removes the json directly without specifying enzyme to json and using toJSON thats provided below
    // all we did was to set up the config.json to run enzyme-to-json over it

    //const renderer = new ReactShallowRenderer();
    //renderer.render(<Header />);
//expect(renderer.getRenderOutput()).toMatchSnapshot();
});

//text() this to access the value of h1


//import toJSON from "enzyme-to-json";
  //expect(toJSON(wrapper)).toMatchSnapshot();