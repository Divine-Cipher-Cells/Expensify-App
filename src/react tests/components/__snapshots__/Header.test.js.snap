// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow test for Header component 1`] = `
<header>
  <h1 />
  <NavLink
    activeClassName="is-active"
    exact={true}
    strict={true}
    to="/"
  >
    Dashboard 
  </NavLink>
  <NavLink
    activeClassName="is-active"
    exact={true}
    strict={true}
    to="/create"
  >
    Create-Expense 
  </NavLink>
</header>
`;
